README.md

```diff
@@ -473,22 +485,193 @@ When the scanner successfully decodes a `TransactionRequest` it will call its de
 </manifest>
 ```
 
+# Websocket
+
+After we are able to create a `TransactionRequest`, consume a `TransactionRequest`, and then approve or reject a `TransactionConsumption`.
+Now, we will learn how to use the web socket client for listening for the `TransactionConsumption` events.
+This section describes the use of the `OMGSocketClient` in order to listen for events.
+
+## Websocket Initialization
+
+Similarly to the `OMGAPIClient`, the `OMGSocketClient` needs to be first initialized with a `OMGSocketClient.Builder` before using it.
+
+```kotlin
+val socketClient = OMGSocketClient.Builder {
+    baseURL = "wss://your_ewallet_base_url/api/socket/"
+    apiKey = YOUR_API_KEY
+    authenticationToken = YOUR_AUTH_TOKEN
+    debug = true
+}.build()
+```
+
+Where:
+* `apiKey` is the API key (typically generated on the admin panel).
+* `authenticationToken` is the token corresponding to an OmiseGO Wallet user retrievable using one of our server-side SDKs.
+> You can find more info on how to retrieve this token in the [OmiseGO server SDK documentations](https://github.com/omisego/ruby-sdk#login). 
+* `baseURL` is the URL of the OmiseGO Wallet API, this needs to be an ws(s) url. **Note**: This need to be ended with '/'.
+* `debug` is a boolean indicating if the SDK should print logs in the console.
```

Comment:
Same remark for the optional+default value part

Reply:
Done